packages/twenty-chrome-extension/src/options/Sidepanel.tsx

```diff
@@ -41,126 +39,106 @@ const StyledActionContainer = styled.div`
 `;
 
 const Sidepanel = () => {
-  const [isAuthenticating, setIsAuthenticating] = useState(false);
   const [isAuthenticated, setIsAuthenticated] = useState(false);
-  const [iframeSrc, setIframeSrc] = useState(
+  const [clientUrl, setClientUrl] = useState(
     import.meta.env.VITE_FRONT_BASE_URL,
   );
-  const [error, setError] = useState('');
-  const [serverBaseUrl, setServerBaseUrl] = useState('');
-  const authenticate = () => {
-    setIsAuthenticating(true);
-    setError('');
-    chrome.runtime.sendMessage(
-      { action: 'launchOAuth' },
-      ({ status, message }) => {
-        if (status === true) {
-          setIsAuthenticated(true);
-          setIsAuthenticating(false);
-          chrome.storage.local.set({ isAuthenticated: true });
-        } else {
-          setError(message);
-          setIsAuthenticating(false);
-        }
-      },
-    );
-  };
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+
+  const setIframeState = useCallback(async () => {
+    const store = await chrome.storage.local.get([
+      'isAuthenticated',
+      'sidepanelUrl',
+      'clientUrl',
+    ]);
+    if (store.isAuthenticated === true) setIsAuthenticated(true);
+
+    if (isDefined(store.sidepanelUrl)) {
+      isDefined(store.clientUrl)
```

Comment:
if / else more readable

Reply:
I think we said with Charles we don't love the `else` pattern and pref to use return/functions when possible